### PR TITLE
Remove codegen tags and add dynamic exports for ROS2 turtlesim

### PR DESCRIPTION
**Pull request by triccyx**
- Remove code generation TAGS
- Add dynamic loading
- Rename and remove unused classes

### CODE REVIEW

**Comment by triccyx** [comment]
The demo is correctly working now.

File: supplementary/alica_ros2/alica_ros2_turtlesim/README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Build the ROS 2 packages using Colcon.
 
 ```
 source /opt/ros/[YOUR DISTRO HERE]/setup.bash
-colcon build alica_ros_turtlesim
+colcon build --continue-on-error --packages-skip alica_ros_proxy alica_tracing alica_ros_turtlesim alica_tests supplementary_tests
```
Comment (Dale-Koenig) [inline supplementary/alica_ros2/alica_ros2_turtlesim/README.md]: Why continue on error?

**Review by Dale-Koenig — APPROVED** [review]

File: supplementary/alica_ros2/alica_ros2_turtlesim/src/base.cpp [hunk 2]
```diff
@@ -90,7 +89,51 @@ Base::~Base()
...
+        std::cerr << "Error:"
+                  << "Lib not exists in this path:" << libraryPath << std::endl;
+        return;
+    } else {
+        std::cerr << "Debug:"
+                  << "Lib exists in this path:" << libraryPath << " for ALICATurtleWorldModelInit" << std::endl;
+    }
+
+    typedef void(InitType)(rclcpp::Node::SharedPtr&, rclcpp::Node::SharedPtr&);
+    std::function<InitType> wminit;
```
Comment (Dale-Koenig) [inline supplementary/alica_ros2/alica_ros2_turtlesim/src/base.cpp]: Actually, this should not be necessary I think.  Maybe can copy what is done in alica turtlesim 1 with respect to world model?  World model doesn't need to be dynamically loaded

**Review by Dale-Koenig — CHANGES_REQUESTED** [review]

File: supplementary/alica_ros2/alica_ros2_turtlesim/README.md (hunk 1 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/README.md]: Ok fixed.

File: supplementary/alica_ros2/alica_ros2_turtlesim/src/base.cpp (hunk 2 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/src/base.cpp]: This is more challenging. I  try to do all the modifications. Note the TurtleSim ROS1  doesn't use anymore the WM

Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/src/base.cpp]: ok

**Comment by triccyx** [comment]
Some actions can still be done:
- Completely remove WM
- Insert class turtle in the global blackboard

File: supplementary/alica_ros2/alica_ros2_turtlesim/src/base_node.cpp [hunk 3]
```diff
@@ -16,7 +15,7 @@ int main(int argc, char** argv)
     std::string name, roleset, master_plan, alica_path;
     int agent_id;
 
-    rclcpp::Node::SharedPtr nh = rclcpp::Node::make_shared("alica_ros2_turtlesim");
+    rclcpp::Node::SharedPtr nh = rclcpp::Node::make_shared("a");
```
Comment (Dale-Koenig) [inline supplementary/alica_ros2/alica_ros2_turtlesim/src/base_node.cpp]: ?

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+install(TARGETS ${PROJECT_NAME}
+  EXPORT ${CMAKE_PROJECT_NAME}Targets
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+)
+
+install(DIRECTORY include/${PROJECT_NAME}/
+  DESTINATION include/${PROJECT_NAME}/
+  FILES_MATCHING PATTERN "*.h" PATTERN "*.hpp"
+  PATTERN ".svn" EXCLUDE
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/CMakeLists.txt]: "A long time ago in a galaxy far, far away...."

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+#include <engine/DefaultUtilityFunction.h>
+#include <engine/UtilityFunction.h>
+
+namespace turtlesim
+{
+class Master : public alica::BasicPlan
+{
+public:
+    Master(alica::PlanContext& context);
+    virtual ~Master();
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h]: I guess all child of `alica::BasicPlan` already has a virtual dtor, therefore it should be override

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+#include <engine/BasicUtilityFunction.h>
+#include <engine/DefaultUtilityFunction.h>
+#include <engine/UtilityFunction.h>
+
+namespace turtlesim
+{
+class Master : public alica::BasicPlan
+{
+public:
+    Master(alica::PlanContext& context);
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h:15]: we have a `create` method and ctor itself, maybe ctor should be private? Or `create` is only for making unique_ptr?

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/turtle.hpp [hunk 7]
```diff
@@ -1,14 +1,20 @@
...
 
 #include <rclcpp/rclcpp.hpp>
 
+#include <functional>
 #include <geometry_msgs/msg/twist.hpp>
+#include <std_msgs/msg/empty.hpp>
 #include <turtlesim/msg/pose.hpp>
 #include <turtlesim/srv/teleport_absolute.hpp>
 
+using namespace std::placeholders;
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/turtle.hpp]: using namespace in headers in general is not a good style

do we really use it here?

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Go2RandomPosition.h [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+public:
+    Go2RandomPosition(alica::BehaviourContext& context);
+    virtual ~Go2RandomPosition();
+    virtual void run();
+    static std::unique_ptr<Go2RandomPosition> create(alica::BehaviourContext& context);
+
+protected:
+    virtual void initialiseParameters();
+
+private:
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Go2RandomPosition.h:16]: seems redundant

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Go2RandomPosition.h [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+#include <boost/dll/alias.hpp>
+#include <engine/BasicBehaviour.h>
+
+namespace turtlesim
+{
+class Go2RandomPosition : public alica::BasicBehaviour
+{
+public:
+    Go2RandomPosition(alica::BehaviourContext& context);
+    virtual ~Go2RandomPosition();
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Go2RandomPosition.h]: `= default`?

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+{
+    if (!instance) {
+        instance = new ALICATurtleWorldModel();
+    }
+}
+
+void ALICATurtleWorldModel::del()
+{
+    delete instance;
+}
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp:24]: instance = nullptr;

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+ALICATurtleWorldModel::ALICATurtleWorldModel()
+{
+    _turtle = std::make_unique<ALICATurtle>(this);
+}
+
+ALICATurtleWorldModel::~ALICATurtleWorldModel() {}
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp]: call del()?

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/turtle.cpp [hunk 12]
```diff
@@ -12,16 +13,27 @@ using namespace std::chrono_literals;
...
+ALICATurtle::ALICATurtle(ALICATurtleWorldModel* wm)
+        : _wm(wm)
+        , spinner(rclcpp::executors::MultiThreadedExecutor(rclcpp::ExecutorOptions(), 1))
 {
-    rclcpp::CallbackGroup::SharedPtr teleportCallback = priv_nh->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
-    priv_nh->get_parameter("name", _name);
-    _priv_nh = priv_nh;
+    _initTrigger = false;
+
+    _priv_nh = rclcpp::Node::make_shared("b");
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/turtle.cpp]: name?

File: supplementary/alica_ros2/alica_ros2_turtlesim/src/base_node.cpp (hunk 3 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/src/base_node.cpp]: Test leftover.
Reverted.

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/turtle.cpp (hunk 12 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/turtle.cpp]: Test leftover.
Reverted.

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Go2RandomPosition.h (hunk 8 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Go2RandomPosition.h:16]: removed

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp (hunk 10 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp:24]: done

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/turtle.hpp (hunk 7 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/turtle.hpp]: removed

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Go2RandomPosition.h (hunk 9 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Go2RandomPosition.h]: Uhm, I think that this constructor can't be defaulted

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h (hunk 5 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h]: Do you mean you prefer `override` or to `remove dtor`?

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h (hunk 6 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h:15]: Not sure it is what we want to have this private.

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/CMakeLists.txt (hunk 4 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/CMakeLists.txt]: :)
done

**Comment by triccyx** [comment]
@Dale-Koenig the last commit is done!

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp (hunk 11 above)
Comment (triccyx) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp]: done

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h (hunk 5 above)
Comment (ntfshard) [reply supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/include/Master.h]: Clang could start complaining about missing override

**Review by Dale-Koenig — APPROVED** [review]

File: supplementary/alica_ros2/alica_ros2_turtlesim/src/base.cpp [hunk 13]
```diff
@@ -1,37 +1,35 @@
...
 
 namespace turtlesim
 {
 
-Base::Base(rclcpp::Node::SharedPtr nh, rclcpp::Node::SharedPtr priv_nh, const std::string& name, const int agent_id, const std::string& roleset,
-        const std::string& master_plan, const std::string& path)
-        : spinner(rclcpp::executors::MultiThreadedExecutor(rclcpp::ExecutorOptions(), 2))
+Base::Base(rclcpp::Node::SharedPtr nh, const std::string& name, const int agent_id, const std::string& roleset, const std::string& master_plan,
+        const std::string& path)
+        : spinner(rclcpp::executors::MultiThreadedExecutor(rclcpp::ExecutorOptions(), 1))
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/src/base.cpp:26]: Curious: 2 -> 1, why?

File: supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp [hunk 14]
```diff
@@ -0,0 +1,36 @@
+#include "world_model.hpp"
+
+namespace turtlesim
+{
+using std::placeholders::_1;
```
Comment (ntfshard) [inline supplementary/alica_ros2/alica_ros2_turtlesim/libalica-ros2-turtlesim/src/world_model.cpp:5]: seems unused

**Review by ntfshard — APPROVED** [review]

**Pull request merged**